add-on/content/serp-fs.js

```diff
@@ -90,17 +97,20 @@ var serpProgressListener = {
               queries.get(domainInfo.reportPrefix)) {
             code = queries.get(domainInfo.reportPrefix);
           }
+          if (aLocation.spec != gLastSearch) {
+            deregisterSerp();
```

Comment:
Could you add a comment here to describe the case where we would want to deregister the SERP, like in the other branches below?

Is this about navigating from one SERP to another? SERP from different engines presumably?

Reply:
Good point, done.